src/include/duckdb/function/table/ordinality_data.hpp

```diff
@@ -0,0 +1,24 @@
+#pragma once
+#include "duckdb/common/types/data_chunk.hpp"
+namespace duckdb {
+struct OrdinalityData {
+	bool with_ordinality = false;
+	idx_t ordinality_column_id;
+	idx_t ord_index = 1;
+	bool ord_reset = false;
+	bool initialized = false;
+
+	void SetOrdinality(DataChunk &chunk, const vector<column_t> &column_ids) {
+		const idx_t ordinality = chunk.size();
+		if (ordinality > 0) {
+			if (ord_reset) {
+				ord_index = 1;
+				ord_reset = false;
+			}
+			D_ASSERT(chunk.data[ordinality_column_id].GetVectorType() == duckdb::VectorType::FLAT_VECTOR);
+			D_ASSERT(chunk.data[ordinality_column_id].GetType().id() == duckdb::LogicalType::INTEGER);
```

Comment:
The .sequence function I use to populate the ordinality column is not defined for `UBIGINT`, so I now use the signed `BIGINT` instead. Please let me know if you specifically want the column to be unsigned, in that case I have to define the `Sequence` function for unsigned integers. I'll consider this resolved unless you tell me otherwise  :+1: